SeaPublicWebsite.Data/DataAccessProvider.cs

```diff
@@ -0,0 +1,38 @@
+﻿using Microsoft.EntityFrameworkCore;
+
+namespace SeaPublicWebsite.Data;
+
+public class DataAccessProvider : IDataAccessProvider
+{
+    private readonly SeaDbContext context;
+
+    public DataAccessProvider(SeaDbContext context)
+    {
+        this.context = context;
+    }
+    
+    public void AddPropertyData(PropertyData propertyData)
+    {
+        context.PropertyData.Add(propertyData);
+        context.SaveChanges();
+    }
+
+    public void UpdatePropertyData(PropertyData propertyData)
+    {
+        context.PropertyData.Update(propertyData);
+        context.SaveChanges();
+    }
+
+    public async Task<PropertyData> GetSinglePropertyData(string reference)
+    {
+        return await context.PropertyData
+            .Include(p => p.Epc)
+            .Include(p => p.PropertyRecommendations)
+            .FirstOrDefaultAsync(p => p.Reference == reference);
```

Comment:
This should be `SingleAsync()`? The calling code checks that the reference is valid, so returning null here seems like a problem